src/pool_tests.rs

```diff
@@ -14,6 +14,13 @@ struct State {
     id: usize,
 }
 
+impl State {
+    fn assert_not_clear(&self) {
```

Comment:
i think this is asserting that the item _is_ cleared (but not dropped); shouldn't it be called
```suggestion
    fn assert_clear(&self) {
```

Reply:
and maybe also consider adding an `assert_not_clear` that does assert `is_cleared` to be false?